packages/web3-common/src/eth_execution_api.ts

```diff
@@ -41,6 +41,7 @@ export interface BaseTransaction {
 	readonly gas: Uint;
 	readonly value: Uint;
 	readonly input: HexStringBytes;
+	readonly chainId: Uint;
```

Comment:
`chainId` isn't apart of the ETH execution spec for transactions sent via RPC, so I don't think it makes sense to add it here

Reply:
Perhaps you would find interesting how ethers.js team dealt with this problem: 

https://github.com/ethers-io/ethers.js/blob/73a46efea32c3f9a4833ed77896a216e3d3752a0/packages/transactions/src.ts/index.ts

Reply:
I think this is correct, I keep forgetting [this](https://eth.wiki/json-rpc/API) spec is outdated/not all inclusive

Reply:
@spacesailor24 From the conversation with the EF team we had on Discord, the impression was that this Wiki is the most accurate specs till now as every client is following it. 

I remember you were doing further research on this topic to conclude it, can you share some reference to what you found? 

@luu-alex Even if we include it into the execution spec, I don't think it will not be a required attribute.

Reply:
So [our docs](https://web3js.readthedocs.io/en/v1.5.2/web3-eth.html#sendtransaction) say (for `web3.eth.sendTransaction`) that `chainId` is optional from the user's perspective. It does [get defaulted](https://github.com/ChainSafe/web3.js/blob/8783f4d64e424456bdc53b34ef1142d0a7cee4d7/packages/web3-eth-accounts/src/index.js#L293) when the user uses `web3.accounts.signTransaction` though

Judging mainly by how 1.x was written, I think we add `chainId` to our spec as optional, as @nazarhussain has suggested. One thing to keep in mind is every property for a _unsigned_ transaction sent to a provider doesn't _have_ to have all properties defined, especially when the signing of that transaction is going to be handled outside of Web3 (e.g. the provider receives the non-exhaustive transaction, fills in missing properties, then signs (think MetaMask))

p.s. It does [look like](https://github.com/ChainSafe/web3.js/blob/ed825d2b260a249b01c3a55c822a9ca2588e961f/packages/web3-eth/src/index.js#L159) a `defaultChain` can be set for the rpc methods, but doesn't seem to have a default value. The [inputTransactionFormatter](https://github.com/ChainSafe/web3.js/blob/ed825d2b260a249b01c3a55c822a9ca2588e961f/packages/web3-core-helpers/src/formatters.js#L202) (which `web3.eth.sendTransaction` uses) also doesn't handle `chainId`, so it looks like a transaction object can be sent to the provider without a `chainId` specified - I think we should confirm this though in 1.x as the final deciding factor for this

Old spec says it's not required, new spec does, but we should be concerned about how 1.x handles it if none of Web3's users seem to have an issue with how it's handling things